### DISCUSSION TITLE
IPSEC Tunnel failed to come up due to "no trusted RSA public key found for ...." for peer (security gateway) cert

### DISCUSSION THREAD

**Discussion by bairathivivek** [strongswan/strongswan#802] [Q&A] [answered]
**System information:**
 - OS: [e.g. Ubuntu 20.04] CentOS-7.8
 - Kernel version (if applicable): [e.g. 5.10] - 3.10
 - strongSwan version(s): [e.g. 5.9.2] - 5.7.2
 - Tested/confirmed with the latest version: [yes/no] No

Issue:
We are bringing up ipsec tunnel between strongswan client and Security gateway (peer). Both have certs signed by the same issuer sub ca. The security gateway has been brought up recently. We are trying to bring up an ipsec tunnel between this new security gateway and strongswan client but we are seeing IKE _AUTH failure sent by strongswan client. The reason showed in strongswan.log is "no trusted RSA public key found for...." received peer cert.

Please see strongswan client log below when IKE_AUTH response is received from security gateway:

[strongswan.log](https://github.com/strongswan/strongswan/files/7659589/strongswan.log)

Dec  3 15:43:06.231 11[ENC] <du_ipsec|1> parsed IKE_AUTH response 1 [ IDr CERT AUTH SA TSi TSr ]
Dec  3 15:43:06.231 11[IKE] <du_ipsec|1> received end entity cert "C=GB, O=XYZSystem Limited, OU=SEGW, CN=Vendor_BKL-WAH"
Dec  3 15:43:06.231 11[IKE] <du_ipsec|1> **no trusted RSA public key found for 'BLK-WAH.xyzsystem.co.uk'**
Dec  3 15:43:06.231 11[ENC] <du_ipsec|1> generating INFORMATIONAL request 2 [ N(AUTH_FAILED) ]
Dec  3 15:43:06.232 11[NET] <du_ipsec|1> sending packet: from 192.168.1.2[500] to 10.10.1.2[500] (80 bytes)

Attaching allcerts.txt (containing all certificates installed on strongswan client), ipsec.conf and strongswan.conf:
[allcerts.txt](https://github.com/strongswan/strongswan/files/7659603/allcerts.txt)
[ipsec.txt](https://github.com/strongswan/strongswan/files/7659600/ipsec.txt)
[strongswan.txt](https://github.com/strongswan/strongswan/files/7659604/strongswan.txt)

We have never seen this kind of issue till now and this scenario has always worked not sure why this is happening now. Is it because of some issue with the certs.
I am using TPM on strongswan client side. Please let me know if this is related to any config issue.

Eagerly waiting for your reply.

Thanks & Regards,
Vivek

**Comment by tobiasbrunner** [comment, marked as answer]
Apparently, the other peer responds with an IDr of `BLK-WAH.xyzsystem.co.uk`. However, that identity is not contained as subjectAltName extension in the end entity certificate received from it. It does contain one for `BKL-WAH.xyzsystem.co.uk`, though (note the switched `K` and `L`). So make sure the peer responds with the correct identity.

**Reply by bairathivivek** [reply]
Thanks for the reply, I missed it somehow. Thank u so much.
